service/apis/meta_service.go

```diff
@@ -105,3 +106,74 @@ func (s MetaService) GetMetadataFields(name string) ([]MetadataField, error) {
 	}
 	return nil, nil
 }
+
+func CheckKeys(keys []string, data map[string]interface{}) error {
+	for k := range keys {
+		if _, ok := data[keys[k]]; !ok {
+			return fmt.Errorf("%q doesn't exist", keys[k])
+		}
+	}
+	return nil
+}
+
+func SetDefaultKey(dkMap map[string]interface{}, data map[string]interface{}) error {
+	for key := range dkMap {
+		//		log.Infof("keys with default %v, %v",key,dkMap[key])
+		if _, ok := data[key]; !ok {
+			data[key] = dkMap[key]
+			log.Infof("%v doesn't exist. set to default %#v", key, dkMap[key])
+		}
+	}
+	return nil
+}
+
+// CreateMetadata save new metadata to the storage
+func (s MetaService) CreateMetadata(meta string, data map[string]interface{}) (map[string]string, error) {
+	var rkeys = []string{"name","fields", "objtype"}
+	err := CheckKeys(rkeys, data)
+	if err != nil {
+		return nil, err
+	}
+/*
+	if data["objtype"] != "metadata" {
+		return nil, errors.New("For metadata create API, Object type should be metadata")
+	}
+	log.Debugf(" %#v ", data["fields"])
+*/
+	fMap := data["fields"].([]interface{})
+	log.Debugf("fMap is %#v",fMap)
+	if len(fMap)>0 {
+		for i := range fMap {
+			rkeys = []string{"fieldName","fieldType"}
+			err = CheckKeys(rkeys, fMap[i].(map[string]interface{}))
+			if err != nil {
+				return nil, err
+			}
+			dkMap := map[string]interface{}{
+				"cardinality":"one",
+				"mandatory": false,
+				"index":false,
+			}
+			err = SetDefaultKey(dkMap,fMap[i].(map[string]interface{}))
+			if fMap[i].(map[string]interface{})["index"].(bool) == true {
+				rkeys = []string{"upsert","tokenizer"}
+				err = CheckKeys(rkeys, fMap[i].(map[string]interface{}))
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
+
+	e := NewEntityService(s.dbclient)
+	uids, err := e.CreateEntity("metadata", data)
+	log.Infof("metadata created/updated: %v", uids)
```

Comment:
if it makes sense to maintain this logging, probably best to move below the err check

Reply:
moved logging